services/datamanager/datasync/sync.go

```diff
@@ -171,14 +169,16 @@ func (s *syncer) SyncDirectory(dir string) {
 						}
 						return
 					}
-					s.syncDataCaptureFile(captureFile)
-				} else {
-					s.syncArbitraryFile(f)
+					return
 				}
-				s.unmarkInProgress(newP)
+				s.syncDataCaptureFile(captureFile)
+			} else {
+				s.syncArbitraryFile(f)
 			}
-		})
-	}
+			s.unmarkInProgress(path)
+		}
+	})
+	<-started
```

Comment:
This one actually wasn't necessary, so removed! And the uploadData one was actually just necessary to ensure that we didn't build the ticker before this returned, because of concurrent access issues with `clock.Ticker` and our `clock.Add` we use in tests. I realized that this was totally accomplishable without a channel by just building the ticker before/outside of the goroutine, so I removed that too.

I added a comment saying why it needs to be built before the goroutine - lmk if you think that's sufficiently explanatory or if it's needs more detail

Reply:
Nice! The code looks much simpler and the comment is very helpful. Thanks!